homework4/WaitAndNotifyExample.java

```diff
@@ -0,0 +1,100 @@
+class Notifyclassbase extends Thread {
+    public void run() {
+        synchronized (this) {
+            System.out.println("Start " + Thread.currentThread().getName());
+            try {
+                System.out.println(Thread.currentThread().getName() + " waiting");
+                this.wait();
+
+            } catch (InterruptedException e) {
+            }
+
+            System.out.println(Thread.currentThread().getName() + " notified");
+            this.notify();
+        }
+    }
+}
+
+class Notify_thread2 extends Thread {
```

Comment:
either use camel-case or snake-case in your code, do not use both
with java, camel-case is used by default, which is the convention followed.